src/commands/index.ts

```diff
@@ -157,9 +157,15 @@ export class CommandDescriptor<Input extends InputKind = InputKind> {
       await state.setMode(Mode.Insert)
     } else if (flags & CommandFlags.SwitchToNormal) {
       await state.setMode(Mode.Normal)
+
+      // Force selection to be non-empty when switching to normal. This is only
+      // necessary because we do not restore selections yet.
+      // TODO: Remove this once https://github.com/71/dance/issues/31 is fixed.
+      state.normalizeSelections(editor)
```

Comment:
This should be in [`Extension.setEditorMode`](https://github.com/71/dance/blob/747fb49e99c93ad8055e39471aff34be8abe70e2/src/extension.ts#L96), otherwise this doesn't happen when the extension is started, for instance.

Reply:
Done.